### DISCUSSION TITLE
[Feature] Shuffle Slideshow Order

### DISCUSSION THREAD

**Discussion by curtwagner1984** [immich-app/immich#4356] [Feature Request]
### The feature

Summary:
Introduce the capability to play slideshows in a randomized order, rather than just chronologically, enhancing the user experience, especially for larger photo collections.

Description:
A common feedback point, even with major platforms like Google Photos, is the absence of an option to display slideshows in a randomized sequence. As evident from [this Google Photos support thread from 2019](https://support.google.com/photos/thread/993155/is-there-a-way-to-have-my-google-photos-shuffle-when-played-as-a-slideshow?hl=en), users are seeking this feature, but it remains unimplemented even today.

The need for such a feature in Immich becomes paramount as it aims to be a robust alternative to Google Photos. Offering features that even industry leaders have overlooked can be a significant differentiator.

Illustration of the Requirement:
Imagine a photo library consisting of 1,000 images, ordered chronologically. With the current system, when initiating a slideshow, especially on platforms like Android TV, users are invariably shown only the most recent additions. To revisit memories from a previous period, users must manually navigate to the older photos and start a slideshow from there, consequently viewing only older memories. For larger photo collections, this can be cumbersome and detracts from the spontaneity and joy of rediscovering memories.

To address this, the proposed enhancement is to introduce a functionality where, upon triggering a slideshow, the next image displayed is randomly chosen from the entire library. Additionally, to provide more focused experiences, it would be useful to enable users to confine this random selection to specific criteria, such as a particular person or a search result.

Existing Work:
I observed [this pull request (#4277)](https://github.com/immich-app/immich/pull/4277) that seems to be a move in this direction. It may serve as a foundation or reference point for this enhancement.

Conclusion:
Randomizing the slideshow sequence adds a dimension of unpredictability and joy to the user experience. By considering this feature, Immich could not only fulfill a longstanding user request but also gain a competitive edge.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by jrasm91** [comment]
I think the linked PR does exactly what you are asking for. Slideshows can be played per album and there is a button to use random order vs next order. When enabled, the next asset is picked randomly.

**Comment by git-computerwizard** [comment]
To further optimize the user experience, we propose improving the shuffle feature within our slideshow functionality to minimize the display of duplicate images. Specifically, we suggest introducing a mechanism that ensures at least 1000 pictures are displayed between each repeated image, thereby reducing the likelihood of seeing the same photo multiple times in succession.

This enhancement is particularly important when dealing with large libraries or albums containing photos organized across multiple subfolders. In current implementations, the shuffle feature can lead to an overwhelming number of duplicate images being shown during a slideshow session, which may detract from the overall enjoyment and engagement the users.

By implementing this improvement, we aim to provide our users with a more varied and enjoyable viewing experience, ultimately enhancing their satisfaction and loyalty to the platform.

**Comment by drix-1** [comment]
Yes, I would love this! Always irrtated me that Google Photos didn't listen and implemented this. It was a feature in Picasa I think, but was dropped when they stopped that project.
